src/main/java/org/hisp/dhis/integration/rapidpro/route/FetchScheduledTrackerEventsRouteBuilder.java

```diff
@@ -0,0 +1,121 @@
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.hisp.dhis.integration.rapidpro.route;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.LoggingLevel;
+import org.hisp.dhis.integration.rapidpro.aggregationStrategy.AttributesAggrStrategy;
+import org.hisp.dhis.integration.rapidpro.aggregationStrategy.ProgramStageEventsAggrStrategy;
+import org.hisp.dhis.integration.rapidpro.aggregationStrategy.TrackedEntityIdAggrStrategy;
+import org.hisp.dhis.integration.rapidpro.processor.FetchDueEventsQueryParamSetter;
+import org.hisp.dhis.integration.rapidpro.processor.SetAttributesEndpointProcessor;
+import org.hisp.dhis.integration.rapidpro.processor.SetProgramStagesHeaderProcessor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Map;
+
+@Component
+public class FetchScheduledTrackerEventsRouteBuilder extends AbstractRouteBuilder
+{
+    @Autowired
+    SetProgramStagesHeaderProcessor setProgramStagesHeaderProcessor;
+
+    @Autowired
+    ProgramStageEventsAggrStrategy programStageEventsAggrStrategy;
+
+    @Autowired
+    FetchDueEventsQueryParamSetter fetchDueEventsQueryParamSetter;
+
+    @Autowired
+    TrackedEntityIdAggrStrategy trackedEntityIdAggrStrategy;
+
+    @Autowired
+    AttributesAggrStrategy attributesAggrStrategy;
+
+    @Autowired
+    SetAttributesEndpointProcessor setAttributesEndpointProcessor;
+
+    @Override
+    protected void doConfigure()
+        throws
+        Exception
+    {
+        from( "servlet:tasks/syncEvents?muteException=true" )
+            .precondition( "{{sync.dhis2.events.to.rapidpro.flows}}" )
+            .removeHeaders( "*" )
+            .to( "direct:fetchAndProcessEvents" )
+            .setHeader( Exchange.CONTENT_TYPE, constant( "application/json" ) )
+            .setBody( constant( Map.of( "status", "success", "data", "Fetched and enqueued due program stage events" ) ) )
+            .marshal().json();
+
+        from( "quartz://fetchDueEvents?cron={{sync.events.schedule.expression:0 0/30 * * * ?}}&stateful=true" )
+            .precondition( "{{sync.dhis2.events.to.rapidpro.flows}}" )
+            .to( "direct:fetchAndProcessEvents" );
+
+        from( "direct:fetchAndProcessEvents" )
+            .routeId( "Fetch And Process Tracker Events" )
+            .to( "direct:fetchDueEvents" )
+            .choice().when( simple( "${exchangeProperty.dueEventsCount} > 0" ) )
+                .split( simple( "${exchangeProperty.dueEvents}" ) )
+                    .marshal().json().transform().body( String.class )
+                    .setHeader( "eventPayload", simple( "${body}" ) )
```

Comment:
Yes. The `eventPayload` will be referenced after the flow has been triggered to update the event status in DHIS2 from `scheduled` to `open`. But as you have pointed out this should ideally be an exchange property rather than a header, so I will change it.